### PR TITLE
Added some new config values and extended the output of deploy:list to be more useful

### PR DESCRIPTION
**Pull request by quiddle**
## URL

Provide `url` in `deploy.json` and `deploy:list` will append the revision key and display the full URL next to each revision key.
## Verbose

If `verbose` is set to true, `deploy:list` will output the relative timestamp, the author and the commit log for the related git hash.

### CODE REVIEW

File: lib/s3-adapter.js [hunk 1]
```diff
@@ -82,12 +85,37 @@ module.exports = Adapter.extend({
...
+      }
 
-    var footer = green('\n\nUse activate() to activate one of these revisions');
+      if (_self.config.url) {
+        revisionMessage += '  (' + blue(_self.config.url + current) + ')';
+      }
+      revisionMessage += '\n';
+
+      if (_self.config.verbose) {
+        var verboseMessage;
```
Comment (danshultz) [inline lib/s3-adapter.js:102]: This should use the taggingAdapter to maintain compatibility `ember-cli-deploy` please.

Comment (quiddle) [inline lib/s3-adapter.js:102]: What part of the tagging adapter? I considered it, but the tagging adapter api only has a createTag function.

**Comment by quiddle** [comment]
So looking at the direction of ember-cli-deploy vs the adapters, how much of this functionality should be moved to ember-cli-deploy via some new methods?

Or some smarts could be moved to the tagging adapter, which could optionally return extended information about a given tag? `taggingAdapter.extendedTagInfo(revision)` or something like that.

Maybe I should move this discussion over to the `ember-cli-deploy` issues?

**Comment by danshultz** [comment]
Ah, sorry - I read the code to quickly. The code is fetching the revision and getting the commit message. Yeah, this is something that ember-cli-deploy would need to be extended to handle. This would be a good discussion for there. Maybe a manifest information plugin or such which could be registered in the plugin registry that can optionally be registered and relied upon.

Please open an issue regarding this in the ember-cli-deploy project.

**Comment by Kerry350** [comment]
Hey @quiddle, just wanted to say thanks for this PR, I really like the extended formatting - great for getting snap overviews of who's done what. I'll leave this open for now, and I'll follow along in the PR over in `ember-cli-deploy` to see what the outcomes are. Great work though :+1:

**Comment by quiddle** [comment]
I really picked  a fun time to get involved in this project! I'll stay on top of things and I'll be happy to contribute changes.

**Comment by quiddle** [comment]
One quick note- I think the formatting is nice (obviously, or I wouldn't have done it), but that can be accomplished in many places. The one thing that will probably need to persist in this adapter is the notion of a `CURRENT` file, if we want to have some way to indicate which revision is currently active.

**Comment by quiddle** [comment]
Rebased to keep these changes in sync. Think I've finally worked that one out.

**Pull request closed without merging**